synthetic_data/dataset_generator.py

```diff
@@ -0,0 +1,89 @@
+
+from synthetic_data.distinct_generators.categorical_generator import random_categorical
+from synthetic_data.distinct_generators.datetime_generator import random_datetimes
+from synthetic_data.distinct_generators.float_generator import random_floats
+from synthetic_data.distinct_generators.int_generator import random_integers
+from synthetic_data.distinct_generators.text_generator import random_string, random_text
+
+
+def convert_data_to_df(
+    np_data: np.array,
+    path: Optional[str] = None,
+    index: bool = False,
+    column_names: Optional[List[str]] = None,
+) -> pd.DataFrame:
+    """
+    Convert np array to a pandas dataframe.
+
+    :param np_data: np array to be converted
+    :type np_data: numpy array
+    :param path: path to output a csv of the dataframe generated
+    :type path: str, None, optional
+    :param index: whether to include index in output to csv
+    :type path: bool, optional
+    :param column_names: The names of the columns of a dataset
+    :type path: List, None, optional
+    :return: a pandas dataframe
+    """
+    # convert array into dataframe
+    if not column_names:
+        column_names = [x for x in range(len(np_data))]
+    dataframe = pd.DataFrame.from_dict(dict(zip(column_names, np_data)))
+    # save the dataframe as a csv file
+    if path:
+        dataframe.to_csv(path, index=index, encoding="utf-8")
+    return dataframe
+
+
+def generate_dataset_by_class(
+    rng: Generator,
+    columns_to_generate: List[dict] = None,
+    dataset_length: int = 100000,
+    path: Optional[str] = None,
+) -> pd.DataFrame:
+    """
+    Randomizes a dataset with a mixture of different data classes.
+
+    :param rng: the np rng object used to generate random values
+    :type rng: numpy Generator
+    :param columns_to_generate: Classes of data to be included in the dataset
+    :type columns_to_generate: List[dict], None, optional
+    :param dataset_length: length of the dataset generated, default 100,000
+    :type dataset_length: int, optional
+    :param path: path to output a csv of the dataframe generated
+    :type path: str, None, optional
+
+    :return: pandas DataFrame
+    """
+    gen_funcs = {
+        "integer": random_integers,
+        "float": random_floats,
+        "categorical": random_categorical,
+        "text": random_text,
+        "datetime": random_datetimes,
+        "string": random_string,
+    }
+
+    if columns_to_generate is None:
+        raise ValueError("columns_to_generate is a required parameter")
+
+    dataset = []
+    column_names = []
+    for col in columns_to_generate:
+        col_ = copy.deepcopy(col)
+        col_generator = col_.pop("generator")
+        if col_generator not in gen_funcs:
+            raise ValueError(f"generator: {col_generator} is not a valid generator.")
+
+        col_generator_function = gen_funcs.get(col_generator)
+        dataset.append(col_generator_function(**col_, num_rows=dataset_length, rng=rng))
```

Comment:
so looking at the generators themselves, it looks this would fail if we have values for `col_`. Essentially, this func is requiring `num_rows=dataset_length, rng=` be the last parameters. whereas, for example,  has a different order.
```python
def random_categorical(
    rng: Generator,
    categories: Optional[List[str]] = None,
    num_rows: int = 1,
    probabilities: Optional[List[float]] = None,
) -> np.array:
```
We probably should have had an abstracted func which enforced the same structure, but we likely need to update this to put rng first, but ensure the other generators have `num_rows` last in the argument order.

Additionally, below, we should validate we don't get errors when we have args for the generator. Right now, `col_` is empty.

Reply:
Some of the other generators do not have num_rows as their last parameter. Should I update those? I just want to make sure I'm not making this PR too convoluted.

Reply:
> Some of the other generators do not have num_rows as their last parameter. Should I update those? I just want to make sure I'm not making this PR too convoluted.

I can understand not wanting to make it too convoluted, but if we annotate the PR in the UI it should simplify the review.

I do think updating to make it last is an ideal situation.